lib/board.rb

```diff
@@ -18,60 +18,54 @@ def print_board_map
         end
     end
 
-    #win conditions use range
 
     def win?(piece)
+
         (0..5).each do |row|
-            # so first index is always 0, so our range is 0..5 for rows [0,1,2,3,4,5]
-            (0..6).each do |col|
-                # meaning our range is 0..6 for columns [0,1,2,3,4,5,6]
+            (0..3).each do |col|
                 return true if @board_map[row][col..col + 3].all? { |cell| cell == piece } #horizontal win
-                # of the row we are on, check set of four by col..col + 3 (same as saying check index 0,1,2,3), we are calling the idices cells, then all method will return true if all cells are the same as the piece
-                
             end
         end
 
-        (0..5).each do |row|
+        (0..2).each do |row|
```

Comment:
had to change some of these range maxes. it was creating an error because it would iterate through something like @board_map[road + i] and go beyond the values that exist and return nil - and you cant index [col] on nil. I made the appropriate changes throughout each win condition